src/components/Articles.jsx

```diff
@@ -11,30 +11,78 @@ const Articles = () => {
     const [isLoading, setIsLoading] = useState(true);
     const [searchParams, setSearchParams] = useSearchParams();
     const [topicQuery, setTopicQuery] = useState(searchParams.get("topic") || "");
-    const [sortByQuery, setSortByQuery] = useState(searchParams.get("sort_by") || "");
-    const [orderQuery, setOrderQuery] = useState(searchParams.get("order") || "");
+    const [sortByQuery, setSortByQuery] = useState(searchParams.get("sort_by") || "created_at");
+    const [orderQuery, setOrderQuery] = useState(searchParams.get("order") || "desc");
+    const [limitQuery, setLimitQuery] = useState(10);
+    const [pageQuery, setPageQuery] = useState(1);
 
     useEffect(() => {
 
         setIsLoading(true);
+        setLimitQuery(10);
+        setPageQuery(1);
 
-        setSearchParams({
-            /* spread ("assign") all current properties in place, then '&&' 
-            returns the key/value pair on the right if the condition on the left is
-            'truthy', in this case whether the query is not an empty string */
-            ...(topicQuery) && { topic: topicQuery },
-            ...(sortByQuery) && { sort_by: sortByQuery },
-            ...(orderQuery) && { order: orderQuery }
+        setSearchParams(() => {
+            return {
+                ...(topicQuery) && { topic: topicQuery },
+                sort_by: sortByQuery,
+                order: orderQuery,
+                limit: limitQuery,
+                p: pageQuery
+            };
         });
 
-        fetchArticles(topicQuery, sortByQuery, orderQuery, )
+        fetchArticles(topicQuery, sortByQuery, orderQuery)
         .then((articlesData) => {
 
             setArticles(articlesData);
             setIsLoading(false);
-        })
+        });
     }, [topicQuery, sortByQuery, orderQuery]);
 
+    useEffect(() => {
+
+        setIsLoading(true);
+        setPageQuery(1);
+        setSearchParams(() => {
+            return {
+                ...(topicQuery) && { topic: topicQuery },
+                sort_by: sortByQuery,
+                order: orderQuery,
+                limit: limitQuery,
+                p: 1
+            };
+        });
+
+        fetchArticles(topicQuery, sortByQuery, orderQuery, limitQuery)
+        .then((articlesData) => {
+
+            setArticles(articlesData);
+            setIsLoading(false);
+        });
+    }, [limitQuery]);
+
+    useEffect(() => {
+
+        setIsLoading(true);
+        setSearchParams(() => {
+            return {
+                ...(topicQuery) && { topic: topicQuery },
+                sort_by: sortByQuery,
+                order: orderQuery,
+                limit: limitQuery,
+                p: pageQuery
+            };
+        });
+
+        fetchArticles(topicQuery, sortByQuery, orderQuery, limitQuery, pageQuery)
+        .then((articlesData) => {
+
+            setArticles(articlesData);
+            setIsLoading(false);
+        });
+    }, [pageQuery]);
+
```

Comment:
There's a lot going on this component with 3 useEffects! It does work, but this isn't very efficient. I would recommend putting some of this logic into its own components with their own useEffects. This isn't urgent, so please feel free to carry on with your other tickets first 😄